paymentsheet/src/main/java/com/stripe/android/link/domain/LinkProminenceFeatureProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.stripe.android.link.domain
+
+import com.stripe.android.core.Logger
+import com.stripe.android.core.utils.FeatureFlags
+import com.stripe.android.link.gate.LinkGate
+import com.stripe.android.paymentsheet.state.LinkState
+import javax.inject.Inject
+
+/**
+ * Provider interface for Link feature prominence settings and behaviors, handling
+ * the logic for determining when to show special Link UI elements or trigger automatic
+ * behaviors to enhance the Link payment experience.
+ */
+internal interface LinkProminenceFeatureProvider {
+
+    /**
+     * On FlowController, upon showing the payment method list, if the user has selects Link, this
+     * method will determine if the 2FA dialog should be shown eagerly and confirmed as selection upon successful
+     * authentication.
+     */
+    fun show2FADialogOnLinkSelectedInFlowController(linkState: LinkState): Boolean
```

Comment:
We wouldn't need to show the 2FA dialog if the user has already authenticated in that same session. Maybe we can rename this?

Reply:
I guessed that was implied here, feels like the naming might be a bit verbose if we include that?   (ie. `show2FADialogIfNeedsVerificationOnLinkSelectedInFlowController`)

Reply:
Simplified naming here to shouldShowEarlyVerificationInFlowController. https://github.com/stripe/stripe-android/pull/10806/commits/b877bc57fe33d9f1cb03eef4d0709af92deaf77e

@tillh-stripe One other improvement can be to remove the `LinkState.LoginState.NeedsVerification` check, since that's not "part of the feature enablement", and do that check directly on the payment selection viewModel.

Reply:
took that condition out of the "feature provider" domain - https://github.com/stripe/stripe-android/pull/10806/commits/175871073ac6b393ec211748b66995f355913ffb

We'll check the verification status along with the feature availability on the payment options side instead, if that sounds good!